iOS/PartySample/app/src/SimpleClient.mm

```diff
@@ -120,12 +86,69 @@ void OnEndLoading(void) override
 };
 
 
+@interface NotificationReceiver : NSObject {
+    id<NotificationReceiver> receiver;
+}
+
+@property(nonatomic, assign) id<NotificationReceiver> receiver;
+
+- (id)init:(id<NotificationReceiver>)receiver;
+
+@end
+
+@implementation NotificationReceiver
+
+@synthesize receiver = _receiver;
+
+- (id)init:(id<NotificationReceiver>)receiver
+{
+    self = [super init];
+    
+    self.receiver = receiver;
+    
+    return self;
+}
+
+- (void)addObservers
+{
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(notificationHandlerDidEnterBackground:)
```

Comment:
>notificationHandlerDidEnterBackground [](http://example.com/codeflow?start=78&length=37)

Do we have to remove this observers?

Reply:
[[NSNotificationCenter defaultCenter] removeObserver:self]; is removing all observers.